File: SuperMessenger/SuperMessenger/Models/GroupModel.cs
using SuperMessenger.Data;
using SuperMessenger.Models.EntityFramework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SuperMessenger.Models
{
    public class GroupModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime CreationDate { get; set; }
        public string ImageName { get; set; }
        public string Type { get; set; }
        public bool IsCreator { get; set; }

        public List<UserInGroupModel> Users { get; set; }
        public List<MessageFileModel> SentFiles { get; set; }
        public List<MessageModel> Messages { get; set; }
        public List<InvitationModel> Invitations { get; set; }
        public List<ApplicationModel> Applications { get; set; }
    }
}

File: SuperMessenger/SuperMessenger/react-client/src/containers/Models/GroupModel.js
export default class GroupModel { 
  constructor(id, name, creationDate, imageName, type, isCreator, usersInGroup, sentFiles, messages, invitations, applications) {
    this.id = id;
    this.name = name;
    this.creationDate = creationDate;
    this.imageName = imageName;
    this.type = type;
    this.isCreator = isCreator;
    this.usersInGroup = usersInGroup;
    this.sentFiles = sentFiles;
    this.messages = messages;
    this.invitations = invitations;
    this.applications = applications;
  }
}